test/acceptance/spec/23-user-activity.js

```diff
@@ -136,6 +136,61 @@ describe('User Activity', () => {
     });
   });
 
+  describe('adding a user filter', () => {
+    describe('returning no audit data', () => {
+      const from = '01/01/1800';
+      const to = '01/01/1900';
+      const user = 'some-chap';
+
+      before(() => {
+        browser.generateReport(from, to, user);
+      });
+
+      it('shows the User Activity page', () => {
+        browser.shouldBeOnUserActivityReport();
+      });
+
+      it('displays an appropriate message advising no data could be found', () => {
+        browser.getText('h2').should.equal(`No usage data for '${user}' found between ${from} and ${to}`);
+      });
+    });
+
+    describe('returning audit data', () => {
+      const from = '23/12/2016';
+      const to = '26/12/2016';
+      const user = 'lev-e2e-tests';
+      let rowHeaders;
+
+      before(() => {
+        browser.generateReport(from, to, user);
+      });
+
+      it('shows the User Activity report page', () => {
+        browser.shouldBeOnUserActivityReport();
+      });
+
+      it('displays an appropriate message including the user filter value', () => {
+        const h2 = `Showing audit data for '${user}' from ${from}, to ${to}`;
+        browser.getText('h2').should.equal(h2);
+        rowHeaders = browser.getText('table.audit > tbody > tr > th');
```

Comment:
Doesn't feel like this belongs in here.

Could we add another `describe` after this block and set this there. (I can see why you don't want it in the previous `before`.) Alternatively, it could go in a try block in the previous `before`.